src/inline.d

```diff
@@ -2154,3 +2183,36 @@ public Expression inlineCopy(Expression e, Scope* sc)
     scope ids = new InlineDoState(sc.parent, null);
     return doInline(e, ids);
 }
+
+/*************************************
+ * Determine if v is 'head const', meaning
+ * that once it is initialized it is not changed
+ * again.
+ *
+ * This is done using a primitive flow analysis.
+ *
+ * v is head const if v is const or immutable.
+ * Otherwise, v is assumed to be head const unless one of the
+ * following is true:
+ *      1. v is a `ref` or `out` variable
+ *      2. v is a parameter and fd is a variadic function
+ *      3. v is assigned to again
+ *      4. the address of v is taken
+ *      5. v is referred to by a function nested within fd
+ *      6. v is ever assigned to a `ref` or `out` variable
+ *      7. v is ever passed to another function as `ref` or `out`
+ *
+ * Params:
+ *      v       variable to check
+ *      fd      function that v is local to
+ * Returns:
+ *      true if v's initializer is the only value assigned to v
+ */
+
+bool isHeadConst(VarDeclaration v, FuncDeclaration fd)
+{
+    if (!v.type.isMutable())
+        return true;            // currently the only case handled atm
+    return false;
```

Comment:
I intend to add tests for cases 1..7 after it, and so the current way of writing it (I.e. early out) is the way it will wind up anyway.